float32[] pointsX
float32[] pointsY
float32[] pointsZ
float32[] colorsR
float32[] colorsG
float32[] colorsB